### PR TITLE
Fix Basilica unintentionally restraining boss mobs

### PR DESCRIPTION
**Pull request by skyleo**
Thanks to Marida from oRO Dev Team for fixing this.
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Stop Basicila from restraining boss monsters.

**Issues addressed:** https://github.com/HerculesWS/Hercules/issues/1276
**Issues partially addressed:** https://github.com/HerculesWS/Hercules/issues/2420

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

File: src/map/battle.c [hunk 1]
```diff
@@ -6612,6 +6608,11 @@ static int battle_check_target(struct block_list *src, struct block_list *target
 	if( (s_bl = battle->get_master(src)) == NULL )
 		s_bl = src;
 
+	if ((flag & BCT_ENEMY) != 0 && (status_get_mode(s_bl) & MD_BOSS) == 0 && (map->getcell(m, src, src->x, src->y, CELL_CHKBASILICA)
```
Comment (Kenpachi2k13) [inline src/map/battle.c]:
```suggestion
	if ((flag & BCT_ENEMY) != 0 && (status_get_mode(s_bl) & MD_BOSS) == 0 && (map->getcell(m, src, src->x, src->y, CELL_CHKBASILICA) != 0
```

File: src/map/battle.c [hunk 2]
```diff
@@ -6612,6 +6608,11 @@ static int battle_check_target(struct block_list *src, struct block_list *target
...
+	    || map->getcell(m, src, target->x, target->y, CELL_CHKBASILICA))) {
```
Comment (Kenpachi2k13) [inline src/map/battle.c]:
```suggestion
	    || map->getcell(m, src, target->x, target->y, CELL_CHKBASILICA) != 0)) {
```

**Comment by skyleo** [comment]
> 
> 
> BTW; You linked this PR under `Issues addressed`. Is this intended?
Nah thx for mentioning.

**Comment by Nurkz** [comment]
Need check then, is that cell from from skill, or script based via *setcell. Or you broke too many scripts based on safecells by cell basilica

**Comment by MishimaHaruna** [comment]
@Nurkz can you show a minimal example of such a script?

**Pull request merged**